File: java/javaproj/src/main/java/org/codereviewpractice/bad/BadEnricher.java
package org.codereviewpractice.bad;

import java.util.*;

public class BadEnricher {
    public void doStuff(List<String> add1, Set<String> add2, boolean flag, String user) {
        for (String k : add1) {
            if (add2.contains(k)) {
                if (flag) {
                    BadGlobals.results.put(k, k.length() * 2);
                } else {
                    BadGlobals.results.put(k, k.length() + 2);
                }
            } else {
                if (user.equals("admin")) {
                    BadGlobals.results.put(k, 999);
                }
            }
        }
    }
}

File: typescript/src/bad/badEnricher.ts
export function badEnricher(add1: string[], add2: Set<string>, flag: boolean, user: string): void {
    for (let key of add1) {
        if (add2.has(key)) {
            if (flag) {
                results.set(key, key.length * 2);
            } else {
                results.set(key, key.length + 2);
            }
        } else {
            if (user === 'admin') {
                results.set(key, 999);
            }
        }
    }
}

